src/sentry/static/sentry/app/stores/globalSelectionStore.jsx

```diff
@@ -34,6 +40,40 @@ const GlobalSelectionStore = Reflux.createStore({
     this.selection = state || getDefaultSelection();
   },
 
+  /**
+   * Initializes the global selection store
+   * If there are query params apply these, otherwise check local storage
+  */
+  loadInitialData(queryParams) {
+    const query = pick(queryParams, Object.values(URL_PARAM));
+    const hasQuery = Object.keys(query).length > 0;
+
+    let globalSelection = getDefaultSelection();
+
+    if (hasQuery) {
+      const parsed = getStateFromQuery(queryParams);
+      globalSelection = {
+        projects: parsed.project || [],
+        environments: parsed.environment || [],
+        datetime: {
+          [DATE_TIME.START]: parsed.start || null,
+          [DATE_TIME.END]: parsed.end || null,
+          [DATE_TIME.PERIOD]: parsed.period || null,
+          [DATE_TIME.UTC]: parsed.utc || undefined,
+        },
+      };
+    } else {
+      try {
+        globalSelection = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
```

Comment:
Should selections be stored per org?

Reply:
Oh, that's a good point, will update this.